registry/txt.go

```diff
@@ -272,6 +291,21 @@ func (pr affixNameMapper) toTXTName(endpointDNSName string) string {
 	return pr.prefix + DNSName[0] + pr.suffix + "." + DNSName[1]
 }
 
+func (pr affixNameMapper) toNewTXTName(endpointDNSName, recordType string) string {
+	DNSName := strings.SplitN(endpointDNSName, ".", 2)
+	recordT := strings.ToLower(recordType + "-")
+
+	// If specified, replace a leading asterisk in the generated txt record name with some other string
+	if pr.wildcardReplacement != "" && DNSName[0] == "*" {
+		DNSName[0] = pr.wildcardReplacement
+	}
+
+	if len(DNSName) < 2 {
+		return pr.prefix + recordT + DNSName[0] + pr.suffix
+	}
+	return pr.prefix + recordT + DNSName[0] + pr.suffix + "." + DNSName[1]
```

Comment:
Hierarchically, I think it would make more sense to have the record type precede the prefix. I personally would like to use:

```
<recordType><prefix><endpointDNSName>
```

Reply:
@haslersn amended